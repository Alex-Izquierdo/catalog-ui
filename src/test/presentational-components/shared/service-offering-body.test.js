import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ServiceOfferingCardBody from '../../../presentational-components/Shared/service-offering-body';

describe('<ServiceOfferingCardBody />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      name: 'Foo',
      display_name: 'display name',
      distributor: 'Red Hat',
      long_description: 'long description'
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<ServiceOfferingCardBody { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with alternative values', () => {
    const wrapper = mount(<ServiceOfferingCardBody name="Foo" description="Bar" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
