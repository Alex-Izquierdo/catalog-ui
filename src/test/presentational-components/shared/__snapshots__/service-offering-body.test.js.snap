// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceOfferingCardBody /> should render correctly 1`] = `
<ServiceOfferingCardBody
  display_name="display name"
  distributor="Red Hat"
  long_description="long description"
  name="Foo"
>
  <CardBody
    className="card-height"
  >
    <div
      className="pf-c-card__body card-height"
    >
      <TextContent>
        <div
          className="pf-c-content"
        >
          <ConditionalLink>
            <Text
              component="h3"
              title="Foo"
            >
              <h3
                className=""
                data-pf-content={true}
                title="Foo"
              >
                <styled.div>
                  <div
                    className="sc-AxjAm fkKIGI"
                  >
                    Foo
                  </div>
                </styled.div>
              </h3>
            </Text>
          </ConditionalLink>
          <Text
            component="small"
          >
            <small
              className=""
              data-pf-content={true}
            >
              Red Hat
               
            </small>
          </Text>
        </div>
      </TextContent>
      <ItemDetails
        long_description="long description"
        toDisplay={
          Array [
            "long_description",
          ]
        }
      >
        <div
          className="card-prop-text"
          key="card-prop-long_description"
        >
          long description
        </div>
      </ItemDetails>
    </div>
  </CardBody>
</ServiceOfferingCardBody>
`;

exports[`<ServiceOfferingCardBody /> should render correctly with alternative values 1`] = `
<ServiceOfferingCardBody
  description="Bar"
  name="Foo"
>
  <CardBody
    className="card-height"
  >
    <div
      className="pf-c-card__body card-height"
    >
      <TextContent>
        <div
          className="pf-c-content"
        >
          <ConditionalLink>
            <Text
              component="h3"
              title="Foo"
            >
              <h3
                className=""
                data-pf-content={true}
                title="Foo"
              >
                <styled.div>
                  <div
                    className="sc-AxjAm fkKIGI"
                  >
                    Foo
                  </div>
                </styled.div>
              </h3>
            </Text>
          </ConditionalLink>
          <Text
            component="small"
          >
            <small
              className=""
              data-pf-content={true}
            >
               
            </small>
          </Text>
        </div>
      </TextContent>
      <ItemDetails
        description="Bar"
        toDisplay={
          Array [
            "description",
          ]
        }
      >
        <div
          className="card-prop-text"
          key="card-prop-description"
        >
          Bar
        </div>
      </ItemDetails>
    </div>
  </CardBody>
</ServiceOfferingCardBody>
`;
